lib/presentation/util/audio_recorder.dart

```diff
@@ -0,0 +1,86 @@
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'dart:io';
+import 'package:audio_waveforms/audio_waveforms.dart';
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+
+class AudioRecorder {
+  final RecorderController recorderController = RecorderController();
+
+  Future<Either<Failure, Success>> startRecordingAudio() async {
+    try {
+      final permission = await recorderController.checkPermission();
+      if (permission) {
+        final tempPath = Directory.systemTemp.path;
+        final currentTime = DateTime.now().millisecondsSinceEpoch;
+        final fileName = 'audio_$currentTime.m4a';
+        await recorderController.record('$tempPath/$fileName');
+        return Right(AudioRecorderStarted());
+      } else {
+        return Left(AudioPermissionDenied());
+      }
+    } catch (exception) {
+      return Left(AudioRecorderFailed());
+    }
+  }
+
+  Future<Either<Failure, AudioRecorderSuccessViewState>>
+      stopRecordingAndSave() async {
+    FileInfo file;
+    List<FileInfo> pickedFiles = [];
+    await recorderController.stop().then((value) {
+      if (value != null) {
+        var recordedFile = File(value);
+        file = FileInfo(recordedFile.path.split('/').last,
+            '${recordedFile.parent.path}/', recordedFile.lengthSync());
+        pickedFiles.add(file);
+      }
+    });
```

Comment:
what happens if error occur here?

Reply:
added error handling now it will display a toast and return to the previous screen